assets/js/modules/reader-revenue-manager/datastore/publications.test.js

```diff
@@ -16,115 +17,202 @@
+					.receiveGetSettings( {} );
 
-				const publications = registry
-					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				expect( fetchMock ).toHaveFetchedTimes( 1 );
-				expect( publications ).toEqual( fixtures.publications );
-				expect( publications ).toHaveLength(
-					fixtures.publications.length
-				);
+				const syncStatus = await registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.syncPublicationOnboardingState();
+
+				expect( syncStatus ).toBeUndefined();
 			} );
 
-			it( 'should not make a network request if publications are already present', async () => {
+			it( 'should update the settings and call the saveSettings endpoint', async () => {
 				registry
 					.dispatch( MODULES_READER_REVENUE_MANAGER )
 					.receiveGetPublications( fixtures.publications );
 
-				const publications = registry
+				const publication = fixtures.publications[ 0 ];
+
+				const settings = {
+					publicationID: publication.publicationId,
+					publicationOnboardingState:
+						PUBLICATION_ONBOARDING_STATES.PENDING_VERIFICATION,
+					publicationOnboardingStateLastSyncedAtMs: 0,
+				};
+
+				fetchMock.post( settingsEndpoint, {
+					body: {
+						...settings,
+						publicationOnboardingState:
+							PUBLICATION_ONBOARDING_STATES.UNSPECIFIED,
+						publicationOnboardingStateLastSyncedAtMs: Date.now(),
+					},
+					status: 200,
+				} );
+
+				registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.receiveGetSettings( settings );
+
+				await registry
+					.dispatch( MODULES_READER_REVENUE_MANAGER )
+					.syncPublicationOnboardingState();
+
+				// Set expectations for settings endpoint.
+				expect( fetchMock ).toHaveFetched( settingsEndpoint );
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+
+				// Set expectations for publication ID.
+				expect(
+					registry
+						.select( MODULES_READER_REVENUE_MANAGER )
+						.getPublicationID()
+				).toEqual( 'ABCDEFGH' );
+
+				// Set expectations for publication onboarding state.
+				expect(
+					registry
+						.select( MODULES_READER_REVENUE_MANAGER )
+						.getPublicationOnboardingState()
+				).toEqual( PUBLICATION_ONBOARDING_STATES.UNSPECIFIED );
+
+				const syncTimeMs = registry
 					.select( MODULES_READER_REVENUE_MANAGER )
-					.getPublications();
-				await untilResolved(
-					registry,
-					MODULES_READER_REVENUE_MANAGER
-				).getPublications();
+					.getPublicationOnboardingStateLastSyncedAtMs();
 
-				expect( fetchMock ).not.toHaveFetched( publicationsEndpoint );
-				expect( publications ).toEqual( fixtures.publications );
-				expect( publications ).toHaveLength(
-					fixtures.publications.length
-				);
+				// Ensure that the sync time is set.
+				expect( syncTimeMs ).not.toBe( 0 );
+
+				// Ensure that date is within the last 5 seconds.
+				expect( syncTimeMs ).toBeLessThanOrEqual( Date.now() );
+
+				expect( syncTimeMs ).toBeGreaterThan( Date.now() - 5000 );
```

Comment:
If we're using the mocked `Date.now()` this can be simplified a little and made more precise:

```suggestion
				expect( syncTimeMs ).toBe( mockNow );

				// Reset the date.
				Date.now = originalDateNow;
```